src/shortcuts/Shortcut.jsx

```diff
@@ -179,22 +177,8 @@ class Shortcut extends Context {
         return true;
     }
 
-    get completionComponent() {
-        switch (this.metadata.subtype) {
-        case "number":
-            console.error(`We don't currently support a completion component for ${this.metadata.subtype}-subtypes; trying to get a completionComponent for ${this.metadata.id}`);
-            return null;
-        case "date":
-            return ContextCalendar;
-        case "multi-choice":
-        case "choice":
-            return ContextListOptions;
-        case "menu":
-            return ContextGetHelp;
-        default:
-            console.error(`We don't currently support a completion component for ${this.metadata.subtype}-subtypes; trying to get a completionComponent for ${this.metadata.id}`);
-            return null;
-        }
+    get completionComponentName() {
+        return this.metadata.subtype;

```

Comment:
Looking for follow up on this.

Reply:
So instead of returning the name, it should return the completion component?

Reply:
I'm not 100% sure if that's the right placement of it, but that's what I was suggesting yes. We can maybe make a task for doing this later? Just not 100% sure where the abstraction should fully lie, but I'm leaning towards the factory being encapsulated at the shortcut level.

Reply:
It's always correct to question switch statements in an OO app! I think the shortcut model is about tracking instances of shortcuts that are related by contexts - typically because the note they are in is open in the editor. Shortcuts should know what type they are and what type of information they store but they are the model and we should keep them that. The separate concern of a UI to set their value should not be handled in the model. The model can say what type of data it contains and then the UI can use a factory class to discover what completion component should be used to complete that shortcut's value. Completion components should be polymorphic to handle a generic interface between them and the rest of the UI as much as possible. We want to keep a class doing one thing and doing it well. I hope that input helps!

Reply:
That's very helpful! Thanks a lot Greg!